colossus/src/main/scala/colossus/service/ServiceClient.scala

```diff
@@ -296,4 +300,13 @@ extends Controller[O,I](codec, ControllerConfig(config.pendingBufferSize, config
     }
   }
 
+
+  override def idleCheck(period: Duration) {
+    super.idleCheck(period)
+
+    if (sentBuffer.size > 0 && sentBuffer.front.isTimedOut(System.currentTimeMillis)) {
+      // the oldest sent message has expired with no response - shutdown the connection
+      disconnect()
```

Comment:
Interesting re: reconnecting - when I was testing I had to explicitly disable reconnection otherwise it did seem to do it. 
The communication aspect is valid - how about I fail that timed out request (and any others) with a timeout error and then close?